ribs/archives/_sliding_boundary_archive.py

```diff
@@ -140,9 +140,14 @@ def __init__(self,
 
         # Sliding boundary specifics.
         self._remap_frequency = remap_frequency
-        self._boundaries = np.full((self._behavior_dim, np.max(self._dims)),
+        # Allocate an extra entry in each row so we can put the upper bound at
+        # the end.
+        self._boundaries = np.full((self._behavior_dim, np.max(self._dims) + 1),
                                    np.inf,
                                    dtype=self.dtype)
+        # Add the upper bounds.
+        for i, dim in enumerate(self._dims):
```

Comment:
Note: This is incorrect since the upper bounds are not fixed. We should add them in during `re_map`.